src/Queue.js

```diff
@@ -1,75 +1,95 @@
-const { SchedulerInterrupt } = require('./constants/index');
+const { SchedulerInterrupt } = require("./constants/index");
 
-// A class representation of a process queue that may hold either a 
+// A class representation of a process queue that may hold either a
 // blocking or non-blocking process
 class Queue {
-    constructor(scheduler, quantum, priorityLevel, queueType) {
-        this.processes = [];
-        // The queue's priority level; the lower the number, the higher the priority
-        this.priorityLevel = priorityLevel;
-        // The queue's parent scheduler
-        this.scheduler = scheduler;
-        // The queue's allotted time slice; each process in this queue is executed for this amount of time in total
-        // This may be done over multiple scheduler iterations
-        this.quantum = quantum;
-        // A counter to keep track of how much time the queue has been executing so far
-        this.quantumClock = 0;
-        this.queueType = queueType;
+  constructor(scheduler, quantum, priorityLevel, queueType) {
+    this.processes = [];
+    // The queue's priority level; the lower the number, the higher the priority
+    this.priorityLevel = priorityLevel;
+    // The queue's parent scheduler
+    this.scheduler = scheduler;
+    // The queue's allotted time slice; each process in this queue is executed for this amount of time in total
+    // This may be done over multiple scheduler iterations
+    this.quantum = quantum;
+    // A counter to keep track of how much time the queue has been executing so far
+    this.quantumClock = 0;
+    this.queueType = queueType;
+  }
+
+  // Enqueues the given process. Return the enqueue'd process
+  enqueue(process) {
+    this.processes.push(process);
+    process.setParentQueue(this);
+    return process;
+  }
+
+  // Dequeues the next process in the queue. Return the dequeue'd process
+  dequeue() {
+    return this.processes.shift();
+  }
+
+  // Return the least-recently added process without removing it from the list of processes
+  peek() {
+    return this.processes[0];
+  }
+
+  isEmpty() {
+    return this.processes.length === 0;
+  }
+
+  getPriorityLevel() {
+    return this.priorityLevel;
+  }
+
+  getQueueType() {
+    return this.queueType;
+  }
+
+  // Manages a process's execution for the given amount of time
+  // Processes that have had their states changed should not be affected
+  // Once a process has received the alloted time, it needs to be dequeue'd and
+  // then handled accordingly, depending on whether it has finished executing or not
+  manageTimeSlice(currentProcess, time) {
```

Comment:
Awesome job on this - I believe you were the first to get this working. I like how simply you handled the exception for processes with mutated states.
Minor quibble - you're using a `let` where by convention you should use `const` (you never reassign the deq variable).